sip-test-kit/src/main/java/de/ikor/sip/foundation/testkit/workflow/whenphase/routeinvoker/impl/FileRouteInvoker.java

```diff
@@ -0,0 +1,77 @@
+package de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.impl;
+
+import static de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.headers.FileExchangeHeaders.*;
+
+import de.ikor.sip.foundation.testkit.workflow.givenphase.Mock;
+import de.ikor.sip.foundation.testkit.workflow.whenphase.routeinvoker.RouteInvoker;
+import java.util.Map;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.*;
+import org.apache.camel.component.file.FileConsumer;
+import org.apache.camel.component.file.FileEndpoint;
+import org.apache.camel.support.EmptyAsyncCallback;
+import org.springframework.stereotype.Component;
+
+/** Invoker class for triggering routes with File consumer */
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class FileRouteInvoker implements RouteInvoker {
+
+  private final CamelContext camelContext;
+
+  @Override
+  public Optional<Exchange> invoke(Exchange inputExchange) {
+    Route route =
+        camelContext.getRoute(
+            (String) inputExchange.getProperty(Mock.ENDPOINT_ID_EXCHANGE_PROPERTY));
```

Comment:
This was extracted to Helper? We can use that helper function now?

Reply:
Yes, refactored according to your comment.